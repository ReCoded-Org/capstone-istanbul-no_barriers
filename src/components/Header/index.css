:root {
  --title-color: #002736;
  --subTitle-color: #000000;
  --background-color: #fc9e01;
  --text-color: #eff4f6;
  --filter-shadow-color: rgba(0, 0, 0, 0.25);
}

.title {
  color: var(--title-color);
  font: "Roboto";
  font-style: normal;
  font-weight: 700;
  margin-bottom: 0;
  font-weight: bold;
  font-size: 4.5rem;
  line-height: 5.25rem;
}

.subTitle {
  color: var(--subTitle-color);
  font: "Roboto";
  font-weight: normal;
  font-size: 1.125rem;
  line-height: 1.3125rem;
  font-style: normal;
  font-weight: 400;
}

.headerImage {
  position: relative;
  width: auto;
}

.headerImageBtn {
  position: absolute;
  top: 1px;
  filter: drop-shadow(0rem 0.25rem 0.25rem var(--filter-shadow-color));
  background-color: var(--background-color);
  color: var(--text-color);
  font-size: 1.375rem;
  padding: 0.3125rem 1.25rem;
  border: none;
  cursor: pointer;
  border-radius: 0.6875rem;
  text-align: center;
}

.headerImageBtn:hover {
  background-color: var(--background-color);
}
