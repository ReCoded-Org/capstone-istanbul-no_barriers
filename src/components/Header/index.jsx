import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";
import { Col, Container, Row } from "react-bootstrap";
import hero from "./hero.png";
import "./index.css";
import { wordsSearchedAction } from "../../redux/action/index";
import { useDispatch } from "react-redux";

const Header = () => {
  const dispatch = useDispatch();
  const { t } = useTranslation();
  const history = useHistory();
  const [input, setInput] = useState("");

  // when it is triggered it set the searchedWords state and it redirects us to the resources page
  const handleFindNowButton = () => {
    dispatch(wordsSearchedAction(input));
    history.push("/findngos");
  };

  return (
    <Container fluid>
      <Row>
        <Col xs={12} className="mt-5 mb-1 header">
          <h1 className="title">{t("homePage.header.title")}</h1>
          <h5 className="subTitle">{t("homePage.header.subtitle")}</h5>
        </Col>
      </Row>
      <Row className="searchers mt-5 d-flex justify-content-around">
        <Col xs={8} md={6}>
          <input
            className="form-control"
            type="text"
            placeholder={t("homePage.header.inputSearch")}
            aria-label="search"
            value={input}
            onChange={(event) => setInput(event.target.value)}
          />
        </Col>
      </Row>
      <Row className="headerImage mt-4 d-flex justify-content-center">
        <Col xs={12} className="d-flex justify-content-center">
          <button
            className="headerImageBtn"
            type="button"
            onClick={handleFindNowButton}
          >
            {t("homePage.header.button")}
          </button>
        </Col>
        <img src={hero} alt="hero" width="80%" className="img-fluid" />
      </Row>
    </Container>
  );
};

export default Header;
