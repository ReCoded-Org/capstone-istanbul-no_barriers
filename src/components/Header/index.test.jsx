import React from "react";
import renderer from "react-test-renderer";
import Header from "./index";
import { Provider } from "react-redux";
import store from "../../redux/store/store";

it("Header renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Header />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
