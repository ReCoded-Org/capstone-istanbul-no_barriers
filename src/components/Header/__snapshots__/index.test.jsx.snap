// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="mt-5 mb-1 header col-12"
    >
      <h1
        className="title"
      >
        homePage.header.title
      </h1>
      <h5
        className="subTitle"
      >
        homePage.header.subtitle
      </h5>
    </div>
  </div>
  <div
    className="searchers mt-5 d-flex justify-content-around row"
  >
    <div
      className="col-lg-8 col-md-3 col-12"
    >
      <input
        aria-label="search"
        className="form-control"
        onChange={[Function]}
        placeholder="homePage.header.inputSearch"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="headerImage mt-4 d-flex justify-content-center row"
  >
    <div
      className="d-flex justify-content-center col-12"
    >
      <button
        className="headerImageBtn"
        onClick={[Function]}
        type="button"
      >
        homePage.header.button
      </button>
    </div>
    <img
      alt="hero"
      src="hero.png"
      width="80%"
    />
  </div>
</div>
`;
